src/motor-operations.cc

```diff
@@ -84,6 +85,95 @@ static char test_acceleration_ok(float acceleration) {
 }
 #endif
 
+struct HistorySlot {
+  int steps[BEAGLEG_NUM_MOTORS];
+  int max_axis_steps;
+};
+
+typedef RingDeque<struct HistorySlot, QUEUE_LEN + 1> SlotHistory;
+typedef FixedArray<unsigned,BEAGLEG_NUM_MOTORS> SlotPartial;
+
+// This class is delegated to handle the update of the absolute
+// position in steps of each motor and it should be callbacked whenever this
+// operations has to be performed.
+class MotionQueueMotorOperations::StatusListener
+                                  : public MotionQueue::PositionChangeCallback {
+public:
+  StatusListener(MotionQueue *backend, MotorCoordSystem *current_position_steps)
+                : backend_(backend),
+                current_position_steps_(current_position_steps) {}
+  virtual void UpdateAbsolutePosition();
+  void PushInHistory(const LinearSegmentSteps &param,
+                       const int defining_axis_steps) {
```

Comment:
this looks like a weird indentation ('const' should be aligned under the 'const' the line above).